rchttp/client.go

```diff
@@ -43,10 +43,25 @@ var DefaultClient = &Client{
 	},
 }
 
+// ClientWithTimeout facilitates creating a client and setting request timeout
+func ClientWithTimeout(timeout time.Duration) (c *Client) {
+	c = DefaultClient
+	c.HTTPClient.Timeout = timeout
+	return c
+}
+
 // Do calls ctxhttp.Do with the addition of exponential backoff
 func (c *Client) Do(ctx context.Context, req *http.Request) (*http.Response, error) {
 	doer := func(args ...interface{}) (*http.Response, error) {
-		return ctxhttp.Do(args[0].(context.Context), args[1].(*http.Client), args[2].(*http.Request))
+		req := args[2].(*http.Request)
```

Comment:
Should the length of the array and the cast be checked?

Reply:
it's not an exported function, so I imagine it wasn't necessary to check internal usage